machine/network/bridge/v1alpha1.go

```diff
@@ -356,42 +360,82 @@ func (service *v1alpha1Network) Update(ctx context.Context, network *networkv1al
 
 // Delete implements kraftkit.sh/api/network/v1alpha1.Delete
 func (service *v1alpha1Network) Delete(ctx context.Context, network *networkv1alpha1.Network) (*networkv1alpha1.Network, error) {
+	embeddedStore, err := store.NewEmbeddedStore[networkv1alpha1.NetworkSpec, networkv1alpha1.NetworkStatus](
+		filepath.Join(
+			config.G[config.KraftKit](ctx).RuntimeDir,
+			"networkv1alpha1",
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var existingNetwork networkv1alpha1.Network
+	err = embeddedStore.Get(ctx, network.Name, storage.GetOptions{}, &existingNetwork)
+	if err != nil && !strings.Contains(err.Error(), "not found") {
+		return nil, err
+	}
```

Comment:
[The store is already wrapping the v1alpha1 implementation](https://github.com/unikraft/kraftkit/blob/staging/machine/network/register_linux.go#L24-L43), meaning that this obfuscates its use.

Reply:
I think the wrapping itself makes everything feel obfuscated, and it's fine. As long as it works correctly >:D